GSA-TTS/FAC
# Upload Page Row Errors Display

# Upload Page Row Errors Display

## Changes:

- On XLSX file upload, schema validation errors send back a list of issues in the format (Row, Column, Filed, Help text, Link). These issues are now displayed as a table in the info box below the file input. 
- The link doesn't go anywhere, so it's currently omitted.
- The audit home page (`/audit/submission-progress`) has all SF-SAC links unlocked. 
- Completing the general information form redirects to the audit home page instead of the federal awards upload page.

## How to test:

Switch to this branch and run normally. Create an audit, if you have none saved locally. Navigate to an upload page, and attempt to upload a spreadsheet. As XLSX is still somewhat under development, you may see things like "Ensure you have uploaded the correct template, or contact an administrator." This happens in cases where a whole field is missing, such as if "auditee_uei" is not included in the template you're using. Once you have everything right (except some row errors), you may see something like this:

![image](https://github.com/GSA-TTS/FAC/assets/91098850/c15826f4-3efd-494a-82fa-465b837a64fb)

That's what we're looking for.
